lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -363,27 +365,245 @@
+    });
+//Something wrong with this code
+    platform.registerFunction(platform.FunctionKind.Array_includes, function (searchElement, fromIndex = undefined) {
+        // ECMAScript 2017 #sec-array.prototype.includes
+
+        if (this === null || this === undefined) {
+            __chakraLibrary.raiseThis_NullOrUndefined("Array.prototype.includes");
+        }
+
+        //Let O be ? ToObject(this value).
+        let o;
+        if (__chakraLibrary.isArray(this)) {
+            o = this;
+        } else {
+            o = __chakraLibrary.Object(this);
+        }
+
+        //Let len be ? ToLength(? Get(O, "length")).
+        //If len is 0, return false.
+        let len = __chakraLibrary.toLength(o["length"]);
+        if (len === 0) {
+            return false;
+        }
+
+        //Let n be ? ToInteger(fromIndex).
+        //Assert: If fromIndex is undefined, then n is 0.
+        let n = __chakraLibrary.toInteger(fromIndex);
+        let k;
+
+        //If n ≥ 0, then
+        //  Let k be n.
+        //Else n < 0,
+        //  Let k be len + n.
+        //  If k < 0, set k to 0.
+        if (n >= 0) {
+            k = n;
+        } 
+        else {
+            k = len + n;
+
+            if (k < 0) {
+                k = 0;
+            }
+        }
+
+        //Repeat, while k < len
+        while (k < len) {
+            //Let elementK be the result of ? Get(O, ! ToString(k)).
+            let elementK = o[k];
+            //If SameValueZero(searchElement, elementK) is true, return true.
+            if ((searchElement === elementK) || (__chakraLibrary.isNaN(searchElement) && __chakraLibrary.isNaN(elementK))) {
+                return true;
+            }
+            //Increase k by 1.
+            k++;
+        }
+        //Return false. 
+        return false;
+    });
+
+    platform.registerFunction(platform.FunctionKind.Array_reduce, function (callbackfn, initialValue = undefined) {
+        // ECMAScript 2017 #sec-array.prototype.reduce
+
+        //Let O be ? ToObject(this value).
+        //Let len be ? ToLength(? Get(O, "length")).
+        let objInfo = __chakraLibrary.CheckArrayAndGetLen(this, "Array.prototype.reduce");
+        let o = objInfo.o;
+        let len = objInfo.len;
+
+        //If IsCallable(callbackfn) is false, throw a TypeError exception.
+        if (typeof callbackfn !== "function") {
+            __chakraLibrary.raiseFunctionArgument_NeedFunction("Array.prototype.reduce");
+        }
+
+        //If len is 0 and initialValue is not present, throw a TypeError exception.
+        if (len === 0 && initialValue === undefined) {
+            __chakraLibrary.raiseEmptyArrayAndInitValueNotPresent("Array.prototype.reduce"); 
+        }
+
+        //Let k be 0.
+        //Let accumulator be undefined.
+        let k = 0;
+        let accumulator = undefined;
+
+        ////If initialValue is present, then
+        ////Set accumulator to initialValue.
+        if (arguments.length > 1) {
```

Comment:
Yes this resolved it.

Reply:
Would it be better to use a rest param for this case?  Using `arguments` in a sloppy-mode function is usually bad news for performance, I thought.

Reply:
rest params may also work -- when discussing this in person, the reason for using arguments was that .reduce behaves differently depending on if the initial value argument is omitted or provided and === undefined. As far as we could tell, this was by spec. I think its totally insane, but here we are.

Reply:
Yeah, I understood the spec requirement; it’s just that `arguments` in sloppy mode is a very strange beast with some really “magic” behavior and implementations (in my experience) will actively try to avoid creating it if they can to reduce overhead.  So I thought (and admittedly this is just a hunch) using a rest param for this would be more JIT/optimizer friendly.